### PR TITLE
samples(storage transfer): add samples and test cases for storage transfer services

### PR DESCRIPTION
**Pull request by mahendra-google**
### Adding samples and tests cases for following storage transfer services:

- [ ] 1. storagetransfer_quickstart
- [ ] 2. storagetransfer_transfer_to_nearline
- [ ] 3. storagetransfer_get_latest_transfer_operation
- [ ] 4. storagetransfer_manifest_request
- [ ] 5. storagetransfer_transfer_from_posix
- [ ] 6. storagetransfer_download_to_posix
- [ ] 7. storagetransfer_transfer_posix_to_posix
- [ ] 8. storagetransfer_create_event_driven_gcs_transfer

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/2824/checks?check_run_id=31994681418) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

Please address this before any reviews.

**Comment by amanda-tarafa** [comment]
Also requesting @JesseLovelace review from a product perspective.

**Comment by amanda-tarafa** [comment]
@JesseLovelace or @danielduhh Can one of you or someone else from your team review these from a product perspective. I'll review from a language perspective after. Thanks.

File: storagetransfer/api/StorageTransfer.Samples/CheckLatestTransferOperationSample.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                string latestOperationName = transferJob.LatestOperationName;
+
+                
+                if (!string.IsNullOrEmpty(latestOperationName))
+                {
+                    _output.WriteLine("The latest operation for transfer job " +jobName+ " is: " +latestOperationName+ "");
+                }
+                else
+                {
+                    _output.WriteLine("Transfer job "+ jobName +" hasn't run yet, try again once after job started running.");
```
Comment (JesseLovelace) [inline storagetransfer/api/StorageTransfer.Samples/CheckLatestTransferOperationSample.cs:59]: This is worded awkwardly, I'd say something more like "try again after the job has started running"

File: storagetransfer/api/StorageTransfer.Samples/CreateEventDrivenGcsTransferSample.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            // Your Google Cloud Project ID
+            string projectId = "my-project-id",
+            // The GCS bucket to transfer data from
+            string sourceBucket = "my-source-bucket",
+            // The GCS bucket to transfer data to
+            string sinkBucket = "my-sink-bucket",
+            // The subscription ID to a Pubsub queue to track
+            string pubSubId = "projects/PROJECT_NAME/subscriptions/SUBSCRIPTION_ID")
+        {
+            // # A useful description for your transfer job
```
Comment (JesseLovelace) [inline storagetransfer/api/StorageTransfer.Samples/CreateEventDrivenGcsTransferSample.cs:39]: Unnecessary # here

**Review by JesseLovelace — APPROVED** [review]
@amanda-tarafa , is the shift from printing to the Console to printing to a test output helper acceptable here?

File: storagetransfer/api/StorageTransfer.Samples/CheckLatestTransferOperationSample.cs (hunk 1 above)
Comment (mahendra-google) [reply storagetransfer/api/StorageTransfer.Samples/CheckLatestTransferOperationSample.cs:59]: I will reword console output

**Comment by mahendra-google** [comment]
> > @amanda-tarafa , is the shift from printing to the Console to printing to an test output helper acceptable here?
> 
> No, definetely not. @mahendra-google please follow contributing documentation both [public](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/main/CONTRIBUTING.md#starting-a-new-sample) and internal. It suggests to use the KMS set of samples as the canonical set of samples to base all other samples on. Thanks.

 @amanda-tarafa  @JesseLovelace I will follow public and internal documentation and KMS set of samples . Also I am going to remove output helper from all files will replace it with Console.Writeline. Thank You.

**Comment by mahendra-google** [comment]
Hi @JesseLovelace I have made code changes as per feedback. Can you please review it again?

**Comment by amanda-tarafa** [comment]
@mahendra-google was this closed by accident?

**Comment by mahendra-google** [comment]
@amanda-tarafa Do I need to re-raise PR from my fork branch?

**Comment by amanda-tarafa** [comment]
I think you can just reopen the PR. The Reopen button is not active for me, but it may be active for you.

If you can't reopen it, create a new one.

We don't have a bot that closes PRs. And it says you closed it. I think you jus accidentally hit the wrong button.

I'll review both PRs, this one and the one for soft delete this week. I'm sorry for the delay, it usually doesn't take me this long to get to these PRs.

**Comment by mahendra-google** [comment]
No worries for delay in reviewing but  I think I did not close PR accidentally  I also came to know when I received an email related to closure.  The same scenario I observed  in restore token soft delete PR as well that was also got closed automatically . I will check for a reopen option if not available I will create new PR. Thank You.

**Comment by mahendra-google** [comment]
@amanda-tarafa  I could not see option to reopen PR. I have created new PR for this (https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/2857)

**Pull request closed without merging**